dodiscover/constraint/skeleton.py

```diff
@@ -0,0 +1,470 @@
+import logging
+from collections import defaultdict
+from itertools import combinations
+from typing import Any, Dict, Iterable, List, Set
+
+import networkx as nx
+import numpy as np
+
+from dodiscover.ci import BaseConditionalIndependenceTest
+
+from ..context import Context
+
+logger = logging.getLogger()
+
+ACCEPTED_MARKOVIAN_SKELETON_METHODS = ["complete", "neighbors", "neighbors_path"]
+
+
+def _iter_conditioning_set(
+    possible_variables: Iterable,
+    x_var: Any,
+    y_var: Any,
```

Comment:
Would it be possible to specific the types here (and at the other places in the code) further? E.g., as numpy array or union of numpy array, pandas dataframe etc.

Reply:
These would be anything that can be a data frame column, so not numpy array, or union of numpy array. I can just constrain the type then for now to what makes sense for us: `str` and `numerics`.

Reply:
This would be the data type within the data frame or numpy array. But here we don't ask for a single float or str, but an array/dataframe. So I was rather thinking of something like
```
x_var: Union[np.ndarray, pd.DataFrame],
y_var: Union[np.ndarray, pd.DataFrame],
```
We don't further specify the data type within the array/frames.

Reply:
See: #25 for the fix. If that is acceptable, I'll add those here too.

Reply:
Added the `Column` type.